### DISCUSSION TITLE
Is there any way to create the tree using the file content instead of using the path?

### DISCUSSION THREAD

**Discussion by AnaytulIslamArpon** [GumTreeDiff/gumtree#313] [Q&A]
the srcTree and destTree are created using the src and dest file path. Is there any way to create the tree using the file content instead of using the path?

`		Tree srcTree = TreeGenerators.getInstance().getTree(src.getAbsolutePath()).getRoot(); // retrieves and applies the default parser for the file 
		Tree destTree = TreeGenerators.getInstance().getTree(dest.getAbsolutePath()).getRoot(); // retrieves and applies the default parser for the file 
`

**Comment by jrfaller** [comment]
Hi!

You can use the `generateFrom()` method to get it from a string or a stream.

Cheers.
